media/CircleEditor/view-sidebar.js

```diff
@@ -1368,51 +1354,53 @@ sidebar.ArgumentView = class {
 
         if (!this.check()) {
-            // TODO Enable posting message
-            // vscode.postMessage({
-            //     command: 'alert',
-            //     text: 'FORMAT ERROR : Please enter commas and numbers only.'
-            // });
+            vscode.postMessage(
+                {command: 'alert', text: 'FORMAT ERROR : Please enter commas and numbers only.'});
             return;
         }
 
         let shape = this._shape.value;
         shape = '{ "data": [' + shape + '] }';
-        shape = JSON.parse(shape).data;
 
-        // const originalString = this._argument._initializer.toString();
-        // const currentString = this._data.value;
+        try {
+            shape = JSON.parse(shape).data;
+        } catch (err) {
+            vscode.postMessage({
+                command: 'alert',
+                text: 'VALIDATION ERROR : Please check your buffer data again.'
+            });
+            return;
+        }
 
         this._editObject._arguments._isChanged = true;
 
-        // TODO will be implement
-        // if (this._argument._initializer && this._editObject._arguments._isChanged) {
-        //     let data;
-        //     if (this._data) {
-        //         data = this._data.value;
-        //     }
-
-        //     const originalType = this._argument._type.dataType;
-
-        //     let result;
-        //     if (data && currentType === originalType && shape ===
-        //     this._argument._type._shape._dimensions ) {
-        //         result = this.editBuffer(data, currentType, shape);
-        //     } else if (data) {
-        //         result = this.changeBufferType(currentType, data, shape);
-        //     }
-
-        //     if (!result) {
-        //         // TODO Enable posting message
-        //         // vscode.postMessage({
-        //         //     command: 'alert',
-        //         //     text: 'VALIDATION ERROR : Please check your buffer data again.'
-        //         // });
-        //         return;
-        //     }
-        // } else {
-        //     this._editObject._arguments._initializer = null;
-        // }
+        if (this._argument._initializer) {
+            let data;
+            if (this._data) {
+                data = this._data.value;
+            }
+
+            const originalType = this._argument._type.dataType;
+
+            let result;
+
+            if (data && currentType === originalType &&
+                shape === this._argument._type._shape._dimensions) {
+                result = this.editBuffer(data, currentType, shape);
+            } else if (data) {
+                result = this.changeBufferType(currentType, data, shape);
+            }
+
+            if (!result) {
+                vscode.postMessage({
+                    command: 'alert',
+                    text: 'VALIDATION ERROR : Please check your buffer data again.'
+                });
+                return;
```

Comment:
`editBuffer` function is returning nothing(void).
And `data` variable can be removed :)

```suggestion
            const originalType = this._argument._type.dataType;

            if (!this._data) {
                result = false;
            } else if (currentType === originalType &&
                shape === this._argument._type._shape._dimensions) {
                this.editBuffer(this._data.value, currentType, shape);
                result = true;
            } else {
                result = this.changeBufferType(currentType, this._data.value, shape);
            }

            if (!result) {
                vscode.postMessage({
                    command: 'alert',
                    text: 'VALIDATION ERROR : Please check your buffer data again.'
                });
                return;
```